cub/test/c2h/custom_type.cuh

```diff
@@ -131,6 +131,11 @@ public:
   {
     return lhs.key == rhs.key && lhs.val == rhs.val;
   }
+
+  friend __host__ __device__ bool operator!=(const CustomType& lhs, const CustomType& rhs)
+  {
+    return !(lhs == rhs);
+  }
```

Comment:
Needed by `CompareDeviceResults`. It also makes sense for a `c2h::custom_type` which is `equal_comparable_t` to also support `!=`.

Reply:
`CompareDeviceResults` is legacy testing facility. Please, compare device vectors using `==` or `thrust::equal`.

Reply:
Even if this adds 2min of compile time to the iterator test?

Reply:
> Even if this adds 2min of compile time to the iterator test?

Yes, I think so. I'd like to get rid of `CompareDeviceResults` as soon as possible. It allocates memory internally, without relying on c2h safe memory allocation we introduced for a reason. Besides that, the less we mask the problem by sticking to legacy solution, the sooner we'll prioritize fixing it.

Reply:
I see. I will change to comparing vectors than.

> the less we mask the problem by sticking to legacy solution, the sooner we'll prioritize fixing it.

You know, I kind of have a half-baked shot at that: #1870 :)